Responsiveness & Accessibility Enhancements for Application Form

stmcpeters opened the pull request:
### Description  
This PR addresses #555 which focuses on making the Techtonica application form fully responsive and accessible across different devices and screen sizes. 

### Changes Made
- [ ] Implement CSS media queries to ensure proper layout adjustments for different devices/screen sizes
- [ ] Improve focus states for keyboard navigation
- [ ] Add necessary ARIA attributes to enhance screen reader compatibility

### Acceptance Criteria  
- [ ] Implement media queries to ensure the form displays well on all screen sizes.
- [ ] Improve accessibility with ARIA attributes, proper focus states, and keyboard navigation.
- [ ] Form is responsive and displays correctly on various screen sizes
- [ ] Submit a PR that merges into the `mvp` branch, not `develop`

### Related Tickets
- Parent Issue:  #389

stmcpeters commented:
> is this a duplicate of [Accessibility Enhancements for Application Form](https://github.com/Techtonica/techtonica.org/pull/615)? If so please close the other PR

hey! this PR #595 handles media queries for mobile and tablet and PR #615 solely focuses on accessibility (ex. adding aria attributes, etc.)

mai-repo reviewed (approved):
@stmcpeters The mobile view looks great -- lgtm! :)

daaimah123 reviewed (approved):
Tested, looks great

stmcpeters commented:
> I'm noticing the form is centered, but the footer is off and slightly to the right. Can you take a look ?
> 
> ![image](https://private-user-images.githubusercontent.com/6878550/419170095-defbf941-ea73-4ea9-9030-67acd594bfef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDExMjQ4NDIsIm5iZiI6MTc0MTEyNDU0MiwicGF0aCI6Ii82ODc4NTUwLzQxOTE3MDA5NS1kZWZiZjk0MS1lYTczLTRlYTktOTAzMC02N2FjZDU5NGJmZWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMDRUMjE0MjIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWJjNThkZGRhNThlYTQ0NDc2MDFmNTM1MDViZGY4NTRmODhiYzY5ZDU3YjU3NWFlNmNhZjhiZGEzMDJlMDdlMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8ULZKCYtwQQke4flx76EfxqChk7HD4S3V6jbMxcV5lE)

I have resolved this issue in [561cf18](https://github.com/Techtonica/techtonica.org/pull/595/commits/561cf186bcc53bb4ff7e3280a95487aa6c8ed9e2) and [a3e84d5](https://github.com/Techtonica/techtonica.org/pull/595/commits/a3e84d5a4a30c096e5559acaa1f0a2d2ca190254) by removing the padding from `.column` from application form templates.

stmcpeters commented:
> Is it possible to make the form wider ? There seems to be a lot of blank space on mobile and the form is squished quite small compared to all the empty space around it
> 
> ![image](https://private-user-images.githubusercontent.com/6878550/419170931-d550177a-b9fd-494c-a633-c88102dd4d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDExMjY5MzksIm5iZiI6MTc0MTEyNjYzOSwicGF0aCI6Ii82ODc4NTUwLzQxOTE3MDkzMS1kNTUwMTc3YS1iOWZkLTQ5NGMtYTYzMy1jODgxMDJkZDRkMmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMDRUMjIxNzE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2EzY2M3ZThiYTVlNTVmN2Y0NWViZDIzOWNkNGFlMjhiZWVjMTNiZGMyZDU3ZDYwMTA4MGFlMzE0NDU2YmNkNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GkFuRdQ1j2H_9yOjBfsTD5bAHWUVlzogAUomMv57q9g)

I have removed extra whitespace surrounding the fieldset form in [d2993b7](https://github.com/Techtonica/techtonica.org/pull/595/commits/d2993b7a9406ee9a4565bd33d89a27b6af77c259), here is the new view on mobile:
<img width="463" alt="Screenshot 2025-03-04 at 2 07 51 PM" src="https://github.com/user-attachments/assets/812f3cd8-ad2a-4d7a-8ce7-0f218d901656" />

derekhouck commented:
I agree with @monikkaelyse's assessment: I'd like to see the mobile view have less horizontal padding. I would argue for completely removing the margin/padding or width restrictions that leads to the blue background being visible on the side for mobile and tablet views. Let's troubleshoot during our meeting this evening.

stmcpeters commented:
### Newest Changes:
- Removed gutters/whitespace in both mobile and tablet views to increase readability
- Applied `box-sizing` styling on all input fields for consistent widths across all screen sizes
- Replaced the dropdown for pronoun selection with checkboxes to allow multiple selections, increasing inclusivity
### Next Steps/Questions:
- Will create a PR to include the new application form styling in Techtonica’s Style Guide Documentation
- Looking for staff feedback on potentially using checkboxes for racial/ethnic identity selection for applicants who identify as two or more races/ethnicities
  - Would long option names create challenges when storing/parsing the data?  
- Currently troubleshooting tooltip positioning issues in mobile/tablet views—open to suggestions! (This could also be separated out into new ticket)
### New Mobile View
https://github.com/user-attachments/assets/67ad1cf3-a4e8-4061-b63d-8d82f7949e56
### New Tablet View
https://github.com/user-attachments/assets/2df813b6-536f-4e3a-b88a-e55f759826ef

daaimah123 reviewed (changes requested):
Some of the dropdown with longer text is unreadable and going off screen for smaller screen sizes, are you able to make the dropdown adjust responsively as well (i.e. gender identity)

stmcpeters commented:
> Some of the dropdown with longer text is unreadable and going off screen for smaller screen sizes, are you able to make the dropdown adjust responsively as well (i.e. gender identity)

Good callout @daaimah123! @monikkaelyse also brought up a similar [issue](https://github.com/Techtonica/techtonica.org/pull/595#issuecomment-2699191930). My mentor, @derekhouck, and I tested different screen sizes in Chrome but couldn’t replicate the issue (attached visuals of what it looks like on my end), so we suspect it might be related to browser default font sizes or the length of the options. Could you provide more details on how to reproduce this, in order to further troubleshoot? 

https://github.com/user-attachments/assets/90386685-495e-4056-97bd-286c6720b02a


https://github.com/user-attachments/assets/ece6fb51-1352-4ac5-926c-22619035af56

derekhouck commented:
> Some of the dropdown with longer text is unreadable and going off screen for smaller screen sizes, are you able to make the dropdown adjust responsively as well (i.e. gender identity)

I suspect this is due to the length of the dropdown options. The OS's native UI is shrinking the font of the options in order to squeeze everything onto the screen. My recommendation would be to convert these fields to radio button or checkbox groups as appropriate.

derekhouck commented:
> My recommendation would be to convert these fields to radio button or checkbox groups as appropriate.

Or reduce the length of the options copy and have definitions and explanations given elsewhere. Perhaps a "Learn More" accordion or a link that opens a new window to an explainer on pronouns.

stmcpeters commented:
Updates:
- [791cdf6](https://github.com/Techtonica/techtonica.org/pull/595/commits/791cdf63331ff6f6181c9b823f93060f24b35ee4) improves the `<select>` element by adding padding and increasing the size of dropdown arrows.
- [9da1e4c](https://github.com/Techtonica/techtonica.org/pull/595/commits/9da1e4cbf4cfd63a918d802a7457fc3d90f56b22) refactors the race/ethnicity selection dropdown by shortening option names and adding a toggled "Learn More" section that provides detailed explanations.
   - also includes more example countries and more inclusive language for options.
Questions/Still To Do:
- i'm considering refactoring the gender selection dropdown to also use a "Learn More" section to shorten options while still providing context. however, I have concerns about simplifying and explaining:
   - Agender/Genderless → One possible solution is to label this as "Agender/Genderless" in the dropdown, with the following explanation in the "Learn More" section:

> Agender/Genderless: Agender people see themselves as neither a man nor a woman, or both. Individuals in this category often describe themselves as genderblank, genderfree, genderless, gendervoid, non-gendered, or null gender.

   - Non-binary/Genderqueer → One possible solution is to consolidate the two Non-binary/Genderqueer options into a single "Non-binary/Genderqueer" category, while mentioning in the "Learn More" section that non-binary people may identify as femme or not femme.
I'd appreciate feedback (@derekhouck @mai-repo @daaimah123 @monikkaelyse) on whether these approaches are inclusive and effective.
- still troubleshooting tooltips overflowing when expanded on mobile and tablet views.

Resources for Gender Explanations:
- [GenderGP](https://www.gendergp.com/what-is-agender/#:~:text=Agender%20Meaning%20and%20Definition&text=Agender%20people%20see%20themselves%20as,as%20gender%2Dfree%20or%20genderless)
- [Human Rights Campaign](https://www.hrc.org/resources/glossary-of-terms)

derekhouck reviewed (approved):
I'm not the best person to comment on specific options or copy for the gender question, but I do whole-heartedly approve of the "Learn More" solution you landed on. I think this will create a much better UX for folks filling out the form. Great job!

mai-repo commented:
@stmcpeters I also like the learn more option.

MichelleGlauser commented:
@stmcpeters I like your solution for the Non-binary/Genderqueer selection and really appreciate the Learn More sections you created. 💗

stmcpeters commented:
Updates:
- [14cb0ff](https://github.com/Techtonica/techtonica.org/pull/595/commits/14cb0ff0bf061cb65ddb5eb811fbfd403336eda2)  makes the entire "Learn More +" clickable instead of just the + icon
- [c0017a4](https://github.com/Techtonica/techtonica.org/pull/595/commits/c0017a49deb1de65151d5e4d693605d9a4dfa942) adds a "Learn More" section to include details for gender identity selection section
- [e555314](https://github.com/Techtonica/techtonica.org/pull/595/commits/e555314da89be573330a9c8456a22a41b503b569) replaces all tooltips with "Learn More" sections for avoid positioning issues on various screen sizes, maintain consistency and improve accessibility

Demo of Changes on Tablet and Mobile Views:

https://github.com/user-attachments/assets/105cc10c-f0b1-4c3e-8069-a33b2f936714



https://github.com/user-attachments/assets/4444778c-e343-44ac-ab23-8db2458f5948





Reviewers: @mai-repo @daaimah123 @monikkaelyse @derekhouck 
- Do the "Learn More" sections improve clarity?
- Are the changes consistent across different screen sizes and devices?
- Confirm that the entire "Learn More +" is clickable
- Do the "Learn More" sections display the correct content without any bugs or glitches?
- Do the "Learn More" sections open/close smoothly and consistently?

mai-repo commented:
@stmcpeters These updates are fantastic—responsive, clean, and well-executed! The "Learn More" feature works seamlessly across all pages and is a brilliant solution to the tooltip issue. Great job troubleshooting and refining the UI—I love this improvement!

stmcpeters commented:
Updates:
I have received approvals from @daaimah123 and other staff to merge this into the `mvp` branch.

Next Steps:
- Create an issue to replace the placeholder text in `app-form-details.html` with accurate content (issue will be linked here when created)
- Create an issue to remove the CSS styling and JS scripts from the templates and moved to the project's main CSS and JS files (issue will be linked here when created)
